### PR TITLE
Add a KOKKOS workflow for GitHub actions and minor updates for the regression tester

### PR DESCRIPTION
**Pull request by ndtrung81**
**Summary**

This PR adds a workflow file for testing the KOKKOS package with the OpenMP backend with selected subfolders.

**Related Issue(s)**

n/a

**Author(s)**

Trung Nguyen (U Chicago)

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

should maintain

**Implementation Notes**

The workflow is executed on a VM with max 4 cores, so the mpiexec command is defined in `config_kokkos.yaml` is `mpiexec -np 2 lmp_binary -k on t 2 -sf kk`. Also, `-pk kokkos newton on neigh half` is specified.

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

### CODE REVIEW

**Comment by akohlmey** [comment]
@ndtrung81 please remember to assign the pull request to me, if you are comfortable with having it merged.

**Comment by ndtrung81** [comment]
@akohlmey thanks for the reminder. I made last minute changes. It is in your hands now. Thanks!

File: .github/workflows/kokkos-regression.yaml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        cmake -S cmake -B build \
+              -C cmake/presets/gcc.cmake \
+              -C cmake/presets/basic.cmake \
+              -D CMAKE_CXX_COMPILER_LAUNCHER=ccache \
+              -D CMAKE_C_COMPILER_LAUNCHER=ccache \
+              -D BUILD_SHARED_LIBS=off \
+              -D DOWNLOAD_POTENTIALS=off \
+              -D PKG_KOKKOS=on \
+              -D Kokkos_ARCH_ARMAVX=on \
+              -D Kokkos_ENABLE_OPENMP=on \
```
Comment (rbberger) [inline .github/workflows/kokkos-regression.yaml]: Would using the kokkos_openmp preset be an option here?

**Review by rbberger — APPROVED** [review]
Assuming having all these tests just run is the priority here, I'm ok with it. Still seeing a lot of failures though. Both in terms of bad input scripts and comparison errors. At some point this has to be fixed and then turned on hot so it actually marks it as failed.

File: .github/workflows/kokkos-regression.yaml (hunk 1 above)
Comment (ndtrung81) [reply .github/workflows/kokkos-regression.yaml]: yes, it is the kokkos_openmp preset.

Comment (ndtrung81) [reply .github/workflows/kokkos-regression.yaml]: @rbberger agreed, the ultimate goal is to have all the failures fixed. The short term goal is to resolve more or less obvious ones (missing installed packages, bad input scripts, timed out runs). Next, we will refine the tolerances for individual thermo quantities (temp, press, energies) for individual test configurations.

File: .github/workflows/kokkos-regression.yaml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        python3 -m pip install numpy pyyaml junit_xml
+        cmake -S cmake -B build \
+              -C cmake/presets/gcc.cmake \
+              -C cmake/presets/basic.cmake \
+              -D CMAKE_CXX_COMPILER_LAUNCHER=ccache \
+              -D CMAKE_C_COMPILER_LAUNCHER=ccache \
+              -D BUILD_SHARED_LIBS=off \
+              -D DOWNLOAD_POTENTIALS=off \
+              -D PKG_KOKKOS=on \
+              -D Kokkos_ARCH_ARMAVX=on \
```
Comment (akohlmey) [inline .github/workflows/kokkos-regression.yaml]: Kokkos_ARCH_ARMAVX??

Comment (ndtrung81) [reply .github/workflows/kokkos-regression.yaml]: sorry, that was my mistake. The tests went through and performance isn't a concern, so it slipped through. I removed this def and switched to using the kokkos-openmp preset.

**Review by stanmoore1 — APPROVED** [review]
@ndtrung81 thanks for working on this. Let me know what issues/bugs you find with testing the KOKKOS package and I'm happy to help get them resolved.

**Pull request merged**